arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowTable.java

```diff
@@ -130,7 +130,16 @@ public Enumerable<Object> query(DataContext root, ImmutableIntList fields,
         // if the split condition has more than two parts it's a binary operator
         // with an additional literal node
         if (data.length > 2) {
-          treeNodes.add(makeLiteralNode(data[2], data[3]));
+          if (data[3].startsWith("decimal")) {
```

Comment:
It seems that string is not a good choice at the moment. I plan to restructure it into Calcite.Sqltype type